src/WebJobs.Script.WebHost/WebScriptHostExceptionHandler.cs

```diff
@@ -48,8 +54,15 @@ public async Task OnTimeoutExceptionAsync(ExceptionDispatchInfo exceptionInfo, T
             // Task ignoreTask = _hostManager.StopAsync();
             // Give the manager and all running tasks some time to shut down gracefully.
             //await Task.Delay(timeoutGracePeriod);
-
-            _applicationLifetime.StopApplication();
+            string workerRuntime = _environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerRuntimeSettingName);
+            if (string.IsNullOrEmpty(workerRuntime) || string.Equals(workerRuntime, RpcWorkerConstants.DotNetLanguageWorkerName, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
@pragnagopa - Discussed this with @mhoeger and null could also mean out of proc in some scenarios, right? If yes, should there be a separate tracking issue where "FunctionWorkerRuntimeSettingName" should be the ultimate source of truth without ambiguities ?

Reply:
Need to rely on `FunctionInvocationDispatcherstate`.  FunctionInvocationDispatcherstate is set to initialized only for out-of-proc.

Reply:
Do not rely on 'workerRuntime' as for a given language it can change out-of-proc or not. For example, when we add a language worker for C#, this is not true.

Reply:
Ah, yes. This would be useless if dotnet goes out-of-proc. Agreed.

Reply:
What if we look for initialized channels that are actively executing the function that timed out and just shut that one down? This relies on the fact that although WebJobs [knows that the function execution has failed](https://github.com/Azure/azure-webjobs-sdk/blob/f7da27d133b9ce0aadf9918770f6ed45748f3c8a/src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs#L291), the RpcWorkerChannel still believes that the invocation is an [executingInvocation](https://github.com/Azure/azure-functions-host/blob/7d449c87e6684cf5cbc2507ba189e3158c9d2ea6/src/WebJobs.Script/Workers/Rpc/RpcWorkerChannel.cs#L43) (maybe that's not a safe assumption in the long-run). And if nothing is found, shut down the whole application? This would help if/when we have multiple language worker processes running

Reply:
@mhoeger I like that idea. Not entirely sure if this would be a much larger enhancement , hmm..

Reply:
This just came up in a CRI analysis. App is using multiple worker processes - 10. Only invocation on one of them timeouted out, rest of the 9 processes are still healthy. What you and @mhoeger proposed makes more sense. It also, makes sense to only restart worker channels that are faulted due to timeout. So introduce, another state for channel 'Faulted' and only restart channels that are faulted.